### PR TITLE
Update Navigation block placeholder

### LINKED ISSUE #23207: Navigation Block: Reconsider placeholder state
When adding a Navigation block to a document, you see the block's placeholder state.

image

There's a number of things that this UI is trying to do, including:
1. Create a new, empty menu.
2. Create a new menu using a sites existing Pages.
3. Choose a previously created menu to populate a new menu.

I think the dropdown UI is a little complicated, and the current copy is awkward to read. There's an [issue to update the copy](https://github.com/WordPress/gutenberg/issues/22987), but it's kind of stalled; The UI is difficult to explain.

Looking at the Image block for inspiration I wound up with something like this:

image

This feels a little easier to understand, but it is still a rather _large_ representation of a real websites navigation. Lets take a look at it in the context **Full Site Editing**, where you would be adding a Navigation block to a page template:

image

After digging through more of the previous Issues and PRs around this subject, I tried rethinking things a little bit; What if we default to adding an _empty_ menu (1 above), with shortcuts for adding existing Pages or Menu's (2 and 3 above):

image

An alternative to this is to default to a menu containing all top-level Pages, with a _somewhat_ hidden option to remove all Link blocks within the menu:

image

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Switch larger standard placeholder to inline menu that fits in more places. This updates the overall placeholder when the block is first inserted. There is a related PR here that updates the empty state when a menu is inserted #26947 

Fixes #23207 

## How has this been tested?

- Add Navigation block both horizontal and vertical
- Add with/without existing Pages
- Add with/without existing Menus
- Confirm adding Pages, Menus, and Links work

## Types of changes

- Switches away from standard Placeholder
- Moves actions from one menu to multiple inline actions

navigation-placeholder

## TODO

- [X] One issue to fix still: the horizontal and vertical type isn't working in editor, but it does work for the menu.

### CODE REVIEW

**Comment by jasmussen** [comment]
Before:

Screenshot 2020-11-23 at 09 58 53

After:

after

This is a massive step in the right direction, but needs some design love. I'm going to push some things to try and get us closer to what Shaun suggests in https://github.com/WordPress/gutenberg/issues/23207#issuecomment-728254260.

**Comment by jasmussen** [comment]
I pushed some tweaks to:

- Create an unselected and selected state of the placeholder.
- Make the unselected state contain skeleton items.
- Polish the selected state.

It now looks like this:

nav

Props to @shaunandrews for the design he created in https://github.com/WordPress/gutenberg/issues/23207#issuecomment-728254260. I think it works really well, and I can think of no better way to try this than in the plugin proper.

It if works well, I would like us to explore how we can make this skeleton setup state a bit more generic, so it's easier to create on a per block basis. It seems like with a little clever component work, all of the unselected/selected, and the white box with a horizontal row of buttons/dropdowns, can be made pretty easily replicable. 

A next step for me is to make this setup state aware of whether the navigation block is horizontal or vertical. Stay tuned.

**Comment by jasmussen** [comment]
Pushed a fix to make the placeholder state orientation aware:

orientation

But I'm not quite feeling it. The shift is rather jarring. I'm going to continue to tinker with this one.

**Comment by jasmussen** [comment]
The placeholder state with semi transparent currentColor works well in other-color themes, btw:

Screenshot 2020-11-23 at 13 56 27

**Comment by jasmussen** [comment]
Alright, I think it's in a pretty good place now:

status

I would appreciate eyes on the code, and good testing!

**Comment by jasmussen** [comment]
Cc @youknowriad @gziolo @karmatosed, I'd appreciate code and design reviews on this as some of our american friends are out for a well deserved break.

**Comment by gziolo** [comment]
I didn't look at the code yet, but my concern is that it too much resembles the navigation menu that I would be fine to see on the frontend. Shouldn't it be more invitational to take some action?

File: packages/block-library/src/navigation/edit.js [hunk 1]
```diff
@@ -78,9 +78,20 @@ function Navigation( {
 		}
 	);
 
+	const blockClassNames = classnames( className, {
```
Comment (gziolo) [inline packages/block-library/src/navigation/edit.js:81]: Nit: As far as I understand `useBlockProps`, you should pass `blockClassNames` as `className` there and you don't have to worry about the class name that block wrapper defines.

**Review by gziolo — COMMENTED** [review]
There are all e2e tests removed related to this block. I would defer to @getdave and @talldan who co-authored them to decide whether it is possible to refactor them to work with the current workflow. It definitely would be beneficial to have user-facing tests to avoid future regressions. Other than that, I don't have any concerns regarding the proposed changes. Nice improvement 💯

File: packages/block-library/src/navigation/edit.js (hunk 1 above)
Comment (mkaz) [reply packages/block-library/src/navigation/edit.js]: Good suggestion, moved classnames to useBlockProps in a97e8f2874

**Comment by talldan** [comment]
Most of the existing end to end tests were for the placeholder, so I can understand they'd need some updating. The new placeholder seems pretty much the same in functionality but different (and a nice improvement) in presentation, so it feels like the tests still have validity.

In the commit description that removed the tests:
> I tried simply bypassing the tests leaving them around to update later,
however using `return true` or commenting out made the linters complain
too much about unused code and defined functions that are not used.

> The file can be restore when needed.

@mkaz My concern with removing the whole file is that while it's easy to bring back, it might just be forgotten, and there's some handy code in the file for mocking the endpoints that will be very useful.

I reckon it'd be nice to leave in place something simpler and `it.skip` the more complex ones if they're challenging to update. While there's an eslint rule against skipping tests, I think it'd be ok for an experimental block, and would just require an eslint-disable. What do you think @gziolo?

**Comment by mkaz** [comment]
> `it.skip`

TIL 😀

**Review by jasmussen — APPROVED** [review]
This is a big step in the right direction, and very worth trying out and following up on.

File: packages/e2e-tests/specs/experiments/navigation.test.js [hunk 2]
```diff
@@ -267,7 +267,7 @@ beforeEach( async () => {
 afterEach( async () => {
 	await setUpResponseMocking( [] );
 } );
-describe( 'Navigation', () => {
+describe.skip( 'Navigation', () => {
```
Comment (getdave) [inline packages/e2e-tests/specs/experiments/navigation.test.js:270]: Is there an Issue created for a follow up here? The code around this is fairly involved and quite easy to break. Would be good to retain some e2e test coverage.

I might be able to help this Friday if folks feel it's worthwhile.

Comment (jasmussen) [reply packages/e2e-tests/specs/experiments/navigation.test.js:270]: I don't know that there's a followup issue, but I think @mkaz meant to look at it soon!

Comment (mkaz) [reply packages/e2e-tests/specs/experiments/navigation.test.js:270]: Tracking issue created here: https://github.com/WordPress/gutenberg/issues/27588

**Pull request merged**